src/datastore/async_methods/loadRelations.js

```diff
@@ -54,6 +54,9 @@ module.exports = function loadRelations (resourceName, instance, relations, opti
             }
           }
 
+          let defKey = def.localKey ? DSUtils.get(instance, def.localKey) : null
```

Comment:
If you implement my suggested change below, then your changes to `loadRelations.js` are unnecessary, and can be removed.